pyomo/contrib/solver/solvers/gurobi/gurobi_direct_base.py

```diff
@@ -0,0 +1,529 @@
+                grb.ITERATION_LIMIT: tc.iterationLimit,
+                grb.NODE_LIMIT: tc.iterationLimit,
+                grb.TIME_LIMIT: tc.maxTimeLimit,
+                grb.SOLUTION_LIMIT: tc.unknown,
+                grb.INTERRUPTED: tc.interrupted,
+                grb.NUMERIC: tc.unknown,
+                grb.SUBOPTIMAL: tc.unknown,
+                grb.USER_OBJ_LIMIT: tc.objectiveLimit,
+            }
+        return GurobiDirectBase._tc_map
+
+    def _postsolve(self, grb_model, solution_loader, has_obj):
+        status = grb_model.Status
+
+        results = Results()
+        results.solution_loader = solution_loader
+        results.timing_info.gurobi_time = grb_model.Runtime
+
+        if grb_model.SolCount > 0:
+            if status == gurobipy.GRB.OPTIMAL:
+                results.solution_status = SolutionStatus.optimal
+            else:
+                results.solution_status = SolutionStatus.feasible
+        else:
+            results.solution_status = SolutionStatus.noSolution
+
+        results.termination_condition = self._get_tc_map().get(
+            status, TerminationCondition.unknown
+        )
+
+        if (
+            results.termination_condition
+            != TerminationCondition.convergenceCriteriaSatisfied
+            and self.config.raise_exception_on_nonoptimal_result
+        ):
+            raise NoOptimalSolutionError()
+
+        if has_obj:
+            try:
+                if math.isfinite(grb_model.ObjVal):
+                    results.incumbent_objective = grb_model.ObjVal
+                else:
+                    results.incumbent_objective = None
+            except (gurobipy.GurobiError, AttributeError):
+                results.incumbent_objective = None
+            try:
+                results.objective_bound = grb_model.ObjBound
+            except (gurobipy.GurobiError, AttributeError):
+                if grb_model.ModelSense == ObjectiveSense.minimize:
+                    results.objective_bound = -math.inf
+                else:
+                    results.objective_bound = math.inf
+        else:
+            results.incumbent_objective = None
+            results.objective_bound = None
+
+        results.extra_info.IterCount = grb_model.getAttr('IterCount')
+        results.extra_info.BarIterCount = grb_model.getAttr('BarIterCount')
+        results.extra_info.NodeCount = grb_model.getAttr('NodeCount')
+
+        self.config.timer.start('load solution')
+        if self.config.load_solutions:
+            if grb_model.SolCount > 0:
+                results.solution_loader.load_vars()
+            else:
+                raise NoFeasibleSolutionError()
+        self.config.timer.stop('load solution')
+
+        # self.config gets copied a the beginning of
+        # solve and restored at the end, so modifying
+        # results.solver_config will not actually
+        # modify self.config
+        results.solver_config = self.config
```

Comment:
We should duplicate the config:
```suggestion
        results.solver_config = config()
```